src/notebook_utils/notebook_utils.py

```diff
@@ -0,0 +1,41 @@
+import itables.options as opt
+from itables import init_notebook_mode, show
+from pandas import DataFrame
+
+
+def display_df(
+        df: DataFrame,
+        layout: dict = None,
+        buttons: list = None,
+        length_menu: list = None
+) -> None:
+    """
+    Display a pandas DataFrame using itables.
```

Comment:
is there a way to do something similar for spark dfs?

Reply:
I added a TODO item. Since spark DF is not loaded in driver's memory before you call something like `show()`. I don't think any of the extension work with spark df. Unless we want to automatically convert spark df to pandas which might overload memory.

Reply:
Are we using the spark pandas api or regular pandas here?

Reply:
> spark pandas api?

Do you mean something like spark_df.toPandas()? `spark_df.toPandas()` returns a regular pandas df.

Reply:
https://towardsdatascience.com/run-pandas-as-fast-as-spark-f5eefe780c45

Reply:
it doesn't work for `df.to_pandas_on_spark()`

Reply:
Ok, so this function should only be used in the case where the data is in a pandas DF, which means it's all in memory and localized to a single machine, e.g. relatively small?

Reply:
yea that's correct. Maybe I should convert first few thousands of records of spark DF if a spark DF is given?

Reply:
I think at the minimum there should be a warning in the function docs about converting spark dfs to pandas dfs, as I assume doing that naively could pull a huge amount of data into memory on a single worker.

If you want to make any enhancements beyond that that's up to you but I'd do them in a different PR. A helper to `head` a spark df into an itable without pulling the rest of the data, where the maximum # of rows is 1K or something seems useful

Reply:
maybe with an offset arg as well

Reply:
👍 added a notice in the doc. I will look into how to properly read a spark df in later PRs.

Reply:
:+1: